src/VisualStudio/Core/Def/Implementation/Library/FindResults/TreeItems/MetadataDefinitionTreeItem.cs

```diff
@@ -36,7 +35,17 @@ public MetadataDefinitionTreeItem(ISymbol definition, ushort glyphIndex)
 
         public override int GoToSource()
         {
-            return VSConstants.E_NOTIMPL;
+            var resolution = _symbolKey.Resolve(_workspace.CurrentSolution.GetCompilationAsync(_referencingProjectId, CancellationToken.None).Result);
```

Comment:
pass into symbolKey.Resolve(),  ignoreAssemblyKey: true since Metadata as source also does that, and to work in typeforwarding scenarios. see if commit `a7e4c1e` is relevant.